core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java

```diff
@@ -110,6 +114,9 @@ public SuggestionBuilder(String name, String suggester) {
             this.suggester = suggester;
         }
 
+        // for use when constructing builders from an input stream
+        protected SuggestionBuilder() { }
+
```

Comment:
I'd try to avoid this, expecially since it looks like the name and suggester field should actually be mandatory. They are used in toXContent and if not set this will cause NPEs. Instead, each implementation of SuggestionBuilder#writeTo() should create the correct instance with name and suggester already read from the stream.